src/utils/memoize.js

```diff
@@ -1,12 +1,13 @@
 /* eslint-disable no-param-reassign */
 export default fn => (...input) => {
   const args = Array.prototype.slice.call(input);
+  const argsToMemo = JSON.stringify(args);
```

Comment:
Can you provide more info about the context of this?  

It is a dangerous default behaviour. First, if you have something that can't be serialized (like functions, DOM elements or anything with circular references) it will crash or fail. Next, it has a performance cost. Ideally, when working with memoize operations, objects should be treated as immutable objects, so that we can only care about the reference. That's not recursive by design. 

You _could_ optionally make it recursive with an additional parameter, but I'm questioning your initial design.

Reply:
Tested it passing function as parameter and nothing actually breaks.

This implementation actually is faster. Comparing strings when we memo e function is faster than comparing array of elements as the current implementation is doing. Here is a link to backup my statement - http://jsben.ch/5AL4M

The reason I did this update is if memo a function which receives object as in param once executed it will always return the memoed result regardless the object. Already tested it.

The way I found the bug and decided and the reason for wanting to fix it is because this function is used when calculating overwrite of colors in theme. This overwrite is doable by passing additional prop `customValues` in theme provider `themeName` object. You can reproduce the issue if you pass customValues once and then try to change the colors again by passing different customValues.

Reply:
As discussed in slack, your performance test is invalid considering that the whole point of your code is to work with complex objects. But I confirm that the original code is not good for complex objects. I still don't think this is a good solution.